auto-discovery/kubernetes/controllers/service_scan_controller.go

```diff
@@ -110,130 +110,133 @@ func (r *ServiceScanReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			RequeueAfter: requeueInterval,
 		}, nil
 	}
-
-	for _, host := range getHostPorts(service) {
-		// Checking if we already have run a scan against this version
-		var scans executionv1.ScheduledScanList
-
-		// construct a map of labels which can be used to lookup the scheduledScan created for this service
-		versionedLabels := map[string]string{
-			"auto-discovery.securecodebox.io/target-service": service.Name,
-			"auto-discovery.securecodebox.io/target-port":    fmt.Sprintf("%d", host.Port),
-			"app.kubernetes.io/managed-by":                   "securecodebox-autodiscovery",
-		}
-		for containerName, podDigest := range podDigests {
-			// The map should only contain one entry at this point. As the reconciler breaks (see containerDigestsAllMatch) if the services points to a list pods with different digests per container name
-			for digest := range podDigest {
-				versionedLabels[fmt.Sprintf("digest.auto-discovery.securecodebox.io/%s", containerName)] = digest[0:min(len(digest), 63)]
-				break
-			}
-		}
-
-		r.Client.List(ctx, &scans, client.MatchingLabels(versionedLabels), client.InNamespace(service.Namespace))
-		log.V(8).Info("Got ScheduledScans for Service in the exact same version", "scheduledScans", len(scans.Items), "service", service.Name, "namespace", service.Namespace)
-
-		if len(scans.Items) != 0 {
-			log.V(8).Info("Service Version was already scanned. Skipping.", "service", service.Name, "namespace", service.Namespace)
-			continue
-		}
-
-		var previousScan executionv1.ScheduledScan
-		err := r.Client.Get(ctx, types.NamespacedName{Name: fmt.Sprintf("%s-service-port-%d", service.Name, host.Port), Namespace: service.Namespace}, &previousScan)
-
-		// generate the scan spec for the current state of the service
-		templateArgs := ServiceAutoDiscoveryTemplateArgs{
-			Config:     r.Config,
-			ScanConfig: r.Config.ServiceAutoDiscoveryConfig.ScanConfig,
-			Cluster:    r.Config.Cluster,
-			Target:     service.ObjectMeta,
-			Service:    service,
-			Namespace:  namespace,
-			Host:       host,
-		}
-		scanSpec := util.GenerateScanSpec(r.Config.ServiceAutoDiscoveryConfig.ScanConfig, templateArgs)
-
-		if apierrors.IsNotFound(err) {
-			// service was never scanned
-			log.Info("Discovered new unscanned service, scanning it now", "service", service.Name, "namespace", service.Namespace)
-
-			versionedLabels = generateScanLabels(versionedLabels, r.Config.ServiceAutoDiscoveryConfig.ScanConfig, templateArgs)
-
-			// No scan for this pod digest yet. Scanning now
-			scan := executionv1.ScheduledScan{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:        fmt.Sprintf("%s-service-port-%d", service.Name, host.Port),
-					Namespace:   service.Namespace,
-					Labels:      versionedLabels,
-					Annotations: generateScanAnnotations(service.Annotations, r.Config.ServiceAutoDiscoveryConfig.ScanConfig, templateArgs),
-				},
-				Spec: scanSpec,
+	for _, scanConfig := range r.Config.ServiceAutoDiscoveryConfig.ScanConfigs {
```

Comment:
I think a warning would be really good here if there are no scanConfigs.
(Had that happen to me as i still had it called `scanConfig`)

```suggestion
	if len(r.Config.ServiceAutoDiscoveryConfig.ScanConfigs) == 0 {
		log.Info("Warning: You have the Service AutoDiscovery enabled but don't have any `scanConfigs` in your AutoDiscovery configuration. No scans will be started!")
	}
	for _, scanConfig := range r.Config.ServiceAutoDiscoveryConfig.ScanConfigs {
```